### DISCUSSION TITLE
All .o files to hidapi-win.zip

### DISCUSSION THREAD

**Discussion by Gnome16** [libusb/hidapi#393] [Ideas]
Just a small request that the Release file hidapi-win.zip also include the hidapi.o files for x64 and x86. I'm writing a Pascal program and would like to statically link it into my program rather than calling the hidapi.dll file externally.

Thanks!

**Comment by Youw** [comment]
`hidapi.dll` has zero external dependencies (except the very basic Windows `dll`s) and can be used with literally any compiler/language on Windows (if it can use PE DLLs at all);
that is the main reason why those provided at all;

`.o` - you likely refering to object files produced by MinGW or Cygwin (note - those are not compatible);
if we are to provide object files (or static libraries) we would have to provide all possible combinations of compiled files: MinGW/Cygwin/MSVC, x86/x64, different compiler runtimes (standard C library), not to mention different threading models for those;

---
Is it feasible? - Absolutely not.

---
If you need static library/object files for your compiler/environment - you're free to compile it on your own.
We have a [readme](https://github.com/libusb/hidapi/blob/master/BUILD.md#integrating-hidapi-directly-into-your-source-tree) with the rescription how to do so.
